### PR TITLE
Fix potential memory leaks in AggregateFunctionForEach

### PR DESCRIPTION
**Pull request by HappenLee**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix potential memory leaks in AggregateFunctionForEach

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Missing test.

**Comment by alexey-milovidov** [comment]
@HappenLee, waiting for tests.

**Comment by alexey-milovidov** [comment]
@HappenLee, could you please add tests?

**Comment by alexey-milovidov** [comment]
@HappenLee, we need tests.

**Comment by alexey-milovidov** [comment]
Hi, @HappenLee! A change requires a test to be accepted.

**Comment by alexey-milovidov** [comment]
@HappenLee, the change needs to be proven by some tests.

**Comment by alexey-milovidov** [comment]
Hi, @HappenLee. How are you? Please add a test.

File: src/AggregateFunctions/Combinators/AggregateFunctionForEach.h [hunk 1]
```diff
@@ -110,6 +110,13 @@ class AggregateFunctionForEach final : public IAggregateFunctionDataHelper<Aggre
 
             state.array_of_aggregate_datas = new_state;
             state.dynamic_array_size = new_size;
+
+            // Since destroyImpl cleans up state.array_of_aggregate_datas. we should to clean up old_states after
```
Comment (alexey-milovidov) [inline src/AggregateFunctions/Combinators/AggregateFunctionForEach.h:114]: "should to" is incorrect.

**Comment by alexey-milovidov** [comment]
@HappenLee, you didn't explain.

**Comment by alexey-milovidov** [comment]
@HappenLee, We need to prove this change with a test.

**Comment by alexey-milovidov** [comment]
@HappenLee We need to know the scenario reproducing the hypothetical problem.

**Comment by alexey-milovidov** [comment]
@HappenLee, we need more information about this change.

**Comment by Algunenano** [comment]
Abandoned

**Pull request closed without merging**